### DISCUSSION TITLE
Youtube-dl requires authentication

### DISCUSSION THREAD

**Discussion by StarEngineer89** [norkunas/youtube-dl-php#250] [Q&A]
Hi,
I use this library in Codeigniter 4.

The code is like here:
$yt = new YoutubeDl();
$yt->setBinPath('C:\ffmpeg\bin\yt-dlp.exe');
$collection = $yt->download(
    Options::create()
        ->downloadPath(WRITEPATH . 'uploads/youtube')
        ->extractAudio(true)
        ->audioFormat('mp3')
        ->audioQuality('0') // best
        ->output($video_id)
        ->url($youtubeLink)
);

But I'm getting this error:
"Error downloading video: [youtube] DpoeBatCwBQ: Sign in to confirm you're not a bot. This helps protect our community. Learn more."

So I tried to use the authenticate method and changed the code like this:
$yt = new YoutubeDl();
$yt->setBinPath('C:\ffmpeg\bin\yt-dlp.exe');
$youtubeUsername = env('youtube_auth_username');
$youtubePassword = env('youtube_auth_password');
$collection = $yt->download(
    Options::create()
        ->authenticate($youtubeUsername, $youtubePassword)
        ->downloadPath(WRITEPATH . 'uploads/youtube')
        ->extractAudio(true)
        ->audioFormat('mp3')
        ->audioQuality('0') // best
        ->output($video_id)
        ->url($youtubeLink)
);

But I'm still getting the same error.
How can I resolve this issue?

Thank you

**Comment by norkunas** [comment]
Hi @StarEngineer89
I didn't encounter this error yet so cannot 100% say how to solve this.
But one solution for you is to try making this work with command line manually and then we'll see how to solve it

**Reply by StarEngineer89** [reply]
Hi, @norkunas 
I run the sample code in the readme.txt with command line manually, but it shows the same error message.

How can I pass this robot check?

Here is the screenshot of this error message.
![image](https://github.com/user-attachments/assets/433bc14b-07a7-4177-b3b2-554b169b0943)

**Reply by norkunas** [reply]
I think https://github.com/norkunas/youtube-dl-php/issues/248 is related to your problem

**Reply by norkunas** [reply]
> I run the sample code in the readme.txt with command line manually, but it shows the same error message.

I meant that you should try to run `yt-dlp ...your arguments for download` directly to see if it works and if something is wrong in this lib

**Reply by StarEngineer89** [reply]
I tried to run this yt-dlp ... command manually in terminal.
But I'm still not sure how to enable oauth2 and link the device.

![image](https://github.com/user-attachments/assets/c016a156-dfdf-448f-8723-4dec8d783509)


> > I run the sample code in the readme.txt with command line manually, but it shows the same error message.
> 
> I meant that you should try to run `yt-dlp ...your arguments for download` directly to see if it works and if something is wrong in this lib

**Reply by norkunas** [reply]
Did you try use cookies from browser or cookies options as warning suggests?

**Reply by StarEngineer89** [reply]
I try to use cookies from browser but I'm not sure what parameter I can use as you can see above screenshot.

**Reply by norkunas** [reply]
All docs are here: https://github.com/yt-dlp/yt-dlp

**Reply by StarEngineer89** [reply]
It still shows this error.
![image](https://github.com/user-attachments/assets/ad30cad9-f6ad-4756-a290-17eaf3039d99)

**Reply by norkunas** [reply]
Then you should ask for help in that repository,as it doesnt work with command line

**Reply by StarEngineer89** [reply]
I tried to use the cookie for the authentication but it still shows another issue.
![image](https://github.com/user-attachments/assets/9279f282-e7ef-42fe-9331-1f5a35699df8)

How can I fix this issue?

**Reply by norkunas** [reply]
signature extraction errors indicates that yt-dlp is out of date

**Comment by StarEngineer89** [comment]


**Reply by norkunas** [reply]
Please don't reply as new answer, but in the same thread :)

**Comment by saintdevera** [comment]
if using the CLI it says: "Sign in to confirm you’re not a bot. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies. Also see  https://github.com/yt-dlp/yt-dlp/wiki/Extractors#exporting-youtube-cookies  for tips on effectively exporting YouTube cookies"
